File: src/main/java/sprykerFileCreator/model/writer/FileWriter.java
package pav.sprykerFileCreator.model.writer;

import com.intellij.openapi.application.Result;
import com.intellij.openapi.command.WriteCommandAction;
import com.intellij.openapi.fileEditor.OpenFileDescriptor;
import com.intellij.openapi.project.Project;
import com.intellij.psi.*;
import com.jetbrains.php.lang.PhpFileType;
import org.jetbrains.annotations.NotNull;

public class FileWriter implements FileWriterInterface {

    private  Project project;

    public FileWriter(Project project) {
        this.project = project;
    }

    @Override
    public PsiElement writeFile(PsiDirectory fileDirectory, String fileName, String phpClassContent) throws Exception {
        PsiFile existingFile = fileDirectory.findFile(fileName + ".php");

        if (existingFile != null) {
            throw new Exception(fileName + " does already exist.");
        }

        PsiFile file = PsiFileFactory
                .getInstance(project)
                .createFileFromText(fileName + ".php", PhpFileType.INSTANCE, phpClassContent);


        new WriteCommandAction(this.project) {

            @Override
            protected void run(@NotNull Result result) throws Throwable {
                fileDirectory.add(file);

                PsiFile virtualFile = fileDirectory.findFile(fileName + ".php");
                if (virtualFile != null) {
                    new OpenFileDescriptor(project, virtualFile.getVirtualFile()).navigate(true);
                }
            }

            @Override
            public String getGroupID() {
                return "Create Command";
            }

        }.execute();

        return file;
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/services/writer/FileWriter.kt
package com.github.tobi812.sprykerplugin.services.writer

import com.intellij.openapi.command.WriteCommandAction
import com.intellij.openapi.fileEditor.OpenFileDescriptor
import com.intellij.openapi.project.Project
import com.intellij.psi.*
import com.jetbrains.php.lang.PhpFileType
import java.lang.Exception

class FileWriter(private val project: Project) : FileWriterInterface {

    @Throws(Exception::class)
    override fun writeFile(fileDirectory: PsiDirectory, fileName: String, phpClassContent: String): PsiElement {
        val existingFile = fileDirectory.findFile("$fileName.php")

        if (existingFile != null) {
            throw Exception("$fileName does already exist.")
        }

        val file = PsiFileFactory
            .getInstance(project)
            .createFileFromText("$fileName.php", PhpFileType.INSTANCE, phpClassContent)

        WriteCommandAction.writeCommandAction(this.project, file)
            .withGroupId("Create Command")
            .run<Throwable> {
                fileDirectory.add(file)
                val virtualFile = fileDirectory.findFile("$fileName.php")
                if (virtualFile != null) {
                    OpenFileDescriptor(project, virtualFile.virtualFile).navigate(true)
                }
            }

        return file
    }

    @Throws(Exception::class)
    override fun createSubdirectory(fileDirectory: PsiDirectory, subDirectoryName: String): PsiDirectory? {
        WriteCommandAction.writeCommandAction(this.project)
            .withGroupId("Create Command")
            .run<Throwable> {
                fileDirectory.createSubdirectory(subDirectoryName)
            }

        return fileDirectory.findSubdirectory(subDirectoryName)
    }
}
